js/options/recorder.js

```diff
@@ -0,0 +1,123 @@
+window.options.recorder = {
+  name: 'recorder',
+
+  state: {
+    start: [true, false],
+    end: [true, false]
+  },
+  run: function () {
+    const div = document.createElement('div')
+    div.innerHTML = `
+    <html>
+    <head>
+   
+    <style>
+    .title {
+      font-family: "Impact", Impact, sans-serif;
+    }
+    .start-button {
+      width: 70px;
+      height: 70px;
+      border: none;
+      border-radius: 100px;
+      outline: none;
+      background: #228b22;
+      color: white; 
+      cursor: pointer;
+      box-shadow: 0 5px 10px rgba(0, 0, 0, 0.15);
+    }
+    .start-button:hover {
+      background: #90EE90;
+    }
+    .stop-button {
+      width: 70px;
+      height: 70px;
+      border: none;
+      border-radius: 100px;
+      outline: none;
+      background: #B22222;
+      color: white; 
+      cursor: pointer;
+      box-shadow: 0 5px 10px rgba(0, 0, 0, 0.15);
+    }
+    .stop-button:hover {
+      background: #FF6347;
+    }
+    </style>
+    </head>
+    <body>
+    <div class = "title" > <b> Screen Recording Buttons: </b> </div>
+    <div class="recorder">
+        <button class="start-button" id="start"> 
+          Start 
+        </button>
+        <button class="stop-button" id="stop" disabled>
+          Stop 
+        </button>
+        </div>  
+
+    <html>
+    
+    `
+    const start = div.querySelector('#start')
+    const stop = div.querySelector('#stop')
+    let mediaRecorder, stream
+
+    start.addEventListener('click', async function () {
+      start.setAttribute('disabled', true)
+      stop.removeAttribute('disabled')
+
+      const stream = await startRecording()
+      const mimeType = 'video/webm'
+      mediaRecorder = createRecorder(stream, mimeType)
+      startRecording()
+    })
+
+    stop.addEventListener('click', () => {
+      stop.setAttribute('disabled', true)
+      start.removeAttribute('disabled')
+      mediaRecorder.stop()
+      stream.getVideoTracks()[0].stop()
+    })
+    async function startRecording () {
+      stream = await navigator.mediaDevices.getDisplayMedia({
+        video: { mediaSource: 'canvas2d' }
```

Comment:
i can see what u're trying to do here, but i don't think this line is actually doing it. is `video { mediaSource }` an actual constraint property that getDisplayMedia knows what do do with? where'd u come across "mediaSource" as a potential property for media constraints? if it is, we might just need to change the value from 'canvas2d' to something else that explicitly specifies the canvas or something